packages/tracing-internal/src/browser/browserTracingIntegration.ts

```diff
@@ -201,88 +185,57 @@ export const browserTracingIntegration = ((_options: Partial<BrowserTracingOptio
-      const { traceId, dsc, parentSpanId, sampled } = propagationContextFromHeaders(sentryTrace, baggage);
-      expandedContext = {
-        traceId,
-        parentSpanId,
-        parentSampled: sampled,
-        ...context,
-        metadata: {
-          // eslint-disable-next-line deprecation/deprecation
-          ...context.metadata,
-          dynamicSamplingContext: dsc,
-        },
-        trimEnd: true,
-      };
-    } else {
-      expandedContext = {
-        trimEnd: true,
-        ...context,
-      };
-    }
+  function _createRouteSpan(client: Client, startSpanOptions: StartSpanOptions): Span | undefined {
+    const { beforeStartSpan, idleTimeout, finalTimeout, childSpanTimeout } = options;
+
+    const isPageloadTransaction = startSpanOptions.op === 'pageload';
 
-    const finalContext: TransactionContext = beforeStartSpan ? beforeStartSpan(expandedContext) : expandedContext;
+    const finalStartSpanOptions: StartSpanOptions = beforeStartSpan
+      ? beforeStartSpan(startSpanOptions)
+      : startSpanOptions;
 
     // If `beforeStartSpan` set a custom name, record that fact
-    finalContext.attributes =
-      finalContext.name !== expandedContext.name
-        ? { ...finalContext.attributes, [SEMANTIC_ATTRIBUTE_SENTRY_SOURCE]: 'custom' }
-        : finalContext.attributes;
-
-    latestRouteName = finalContext.name;
-    if (finalContext.attributes) {
-      latestRouteSource = finalContext.attributes[SEMANTIC_ATTRIBUTE_SENTRY_SOURCE];
-    }
+    const attributes = finalStartSpanOptions.attributes || {};
 
-    if (finalContext.sampled === false) {
-      DEBUG_BUILD && logger.log(`[Tracing] Will not send ${finalContext.op} transaction because of beforeNavigate.`);
+    if (finalStartSpanOptions.name !== finalStartSpanOptions.name) {
+      attributes[SEMANTIC_ATTRIBUTE_SENTRY_SOURCE] = 'custom';
+      finalStartSpanOptions.attributes = attributes;
     }
 
-    DEBUG_BUILD && logger.log(`[Tracing] Starting ${finalContext.op} transaction on scope`);
+    latestRouteName = finalStartSpanOptions.name;
+    latestRouteSource = attributes[SEMANTIC_ATTRIBUTE_SENTRY_SOURCE];
 
-    const { location } = WINDOW;
+    DEBUG_BUILD && logger.log(`[Tracing] Starting ${finalStartSpanOptions.op} idle span on scope`);
 
-    const idleTransaction = startIdleTransaction(
-      hub,
-      finalContext,
+    const idleSpan = startIdleSpan(finalStartSpanOptions, {
       idleTimeout,
       finalTimeout,
-      true,
-      { location }, // for use in the tracesSampler
-      heartbeatInterval,
-      isPageloadTransaction, // should wait for finish signal if it's a pageload transaction
-    );
+      childSpanTimeout,
+      // should wait for finish signal if it's a pageload transaction
+      disableAutoFinish: isPageloadTransaction,
+      beforeSpanEnd: span => {
+        _collectWebVitals();
+        addPerformanceEntries(span);
+      },
+    });
+
+    if (!idleSpan) {
+      return;
+    }
```

Comment:
Shouldn't `idleSpan` always be defined? I think this would also allow us to change the return type of `_createRouteSpan` to always return a `Span`.

Reply:
actually yes! forgot to update this (started the PR before I made the other changes...) will update this 👍